core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala

```diff
@@ -1206,6 +1206,67 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with B
     }
   }
 
+  test("successful tasks from previous attempts could be learnt by later active taskset") {
```

Comment:
this test is good, but could it instead be folded into "Completions in zombie tasksets update status of non-zombie taskset"?  that would make it easier to maintain (if you think there is a reason it needs to be separate, feel free to push back)

Reply:
Good idea!